application/account-management/Tests/Api/Users/UserEndpointsTests.cs

```diff
@@ -74,6 +74,88 @@ public async Task GetUser_WhenInvalidUserId_ShouldReturnBadRequest()
         await EnsureErrorStatusCode(response, HttpStatusCode.BadRequest, $"""Failed to bind parameter "UserId Id" from "{invalidUserId}".""");
     }
     
+    [Fact]
+    public async Task GetUsers_WhenSearchingBasedOnUserEmail_ShouldReturnUser()
+    {
+        // Arrange
+        var searchEmail = string.Concat(DatabaseSeeder.User1ForSearching.Email.Take(4));
+        
+        // Act
+        var response = await TestHttpClient.GetAsync($"/api/users?search={searchEmail}");
+        
+        // Assert
+        EnsureSuccessGetRequest(response);
+        var userResponse = await DeserializeResponse<GetUsersResponseDto>(response);
+        userResponse.Should().NotBeNull();
+        userResponse!.TotalCount.Should().Be(1);
+        userResponse.Users.First().Email.Should().Be(DatabaseSeeder.User1ForSearching.Email);
+    }
+    
+    [Fact]
+    public async Task GetUsers_WhenSearchingBasedOnUserFirstName_ShouldReturnUser()
+    {
+        // Arrange
+        var searchEmail = string.Concat(DatabaseSeeder.User1ForSearching.FirstName!.Take(4));
+        
+        // Act
+        var response = await TestHttpClient.GetAsync($"/api/users?search={searchEmail}");
+        
+        // Assert
+        EnsureSuccessGetRequest(response);
+        var userResponse = await DeserializeResponse<GetUsersResponseDto>(response);
+        userResponse.Should().NotBeNull();
+        userResponse!.TotalCount.Should().Be(1);
+        userResponse.Users.First().FirstName.Should().Be(DatabaseSeeder.User1ForSearching.FirstName);
+    }
+    
+    [Fact]
+    public async Task GetUsers_WhenSearchingBasedOnUserLastName_ShouldReturnUser()
+    {
+        // Arrange
+        var searchEmail = string.Concat(DatabaseSeeder.User1ForSearching.LastName!.Take(4));
+        
+        // Act
+        var response = await TestHttpClient.GetAsync($"/api/users?search={searchEmail}");
+        
+        // Assert
+        EnsureSuccessGetRequest(response);
+        var userResponse = await DeserializeResponse<GetUsersResponseDto>(response);
+        userResponse.Should().NotBeNull();
+        userResponse!.TotalCount.Should().Be(1);
+        userResponse.Users.First().LastName.Should().Be(DatabaseSeeder.User1ForSearching.LastName);
+    }
+    
+    [Fact]
+    public async Task GetUsers_WhenSearchingBasedOnUserRole_ShouldReturnUser()
+    {
+        // Arrange
+        // Act
+        var response = await TestHttpClient.GetAsync($"/api/users?userRole={UserRole.TenantUser}");
+        
+        // Assert
+        EnsureSuccessGetRequest(response);
+        var userResponse = await DeserializeResponse<GetUsersResponseDto>(response);
+        userResponse.Should().NotBeNull();
+        userResponse!.TotalCount.Should().Be(1);
+        userResponse.Users.First().UserRole.Should().Be(DatabaseSeeder.User1ForSearching.UserRole);
+    }
+    
+    [Fact]
+    public async Task GetUsers_WhenSearchingWithSpecificOrdering_ShouldReturnOrderedUsers()
+    {
+        // Arrange
```

Comment:
For consistency with other tests, leave out the `// Arrange` comment if there are no arrange part.